upload/catalog/model/extension/shipping/fedex.php

```diff
@@ -177,11 +177,11 @@ function getQuote($address) {
 				foreach ($rate_reply_details as $rate_reply_detail) {
 					$code = strtolower($rate_reply_detail->getElementsByTagName('ServiceType')->item(0)->nodeValue);
 
-					if (in_array(strtoupper($code), $this->config->get('shipping_fedex_service'))) {
+						if (is_array($this->config->get('shipping_fedex_service')) && in_array(strtoupper($code), $this->config->get('shipping_fedex_service'))) {
```

Comment:
```if (!empty($this->config->get('shipping_fedex_service')) && is_array($this->config->get('shipping_fedex_service')) && in_array(strtoupper($code), $this->config->get('shipping_fedex_service'))) {```

Reply:
Code added

Reply:
Take note that is_array does not need to be used since the new restructuration of array instantiation in PHP 7 as well as the conversion in Opencart to no longer use (array) but [] instead. By removing the is_array validation, you could still validate the $this->config->get('shipping_fedex_service') as an array in the in_array function instead. This will shorten the codes a bit.